### PR TITLE
fix: capture init log errors

### PR DESCRIPTION
**Pull request by vsukhin**
## Pull request description 

Capture init log errors

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [x] tested locally
- [x] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: pkg/jobs/jobclient.go [hunk 1]
```diff
@@ -377,79 +377,111 @@ func (c *JobClient) GetLastLogLineError(ctx context.Context, podNamespace, podNa
...
-	if err != nil {
-		return logs, err
+		podLogRequest := c.ClientSet.CoreV1().
+			Pods(c.Namespace).
+			GetLogs(pod.Name, &podLogOptions)
+
+		stream, err := podLogRequest.Stream(context.TODO())
+		if err != nil {
+			if len(logs) != 0 {
+				return logs, nil
```
Comment (vLia) [inline pkg/jobs/jobclient.go:409]: is there a way to check for the type of error before returning nil?

File: pkg/jobs/jobclient.go [hunk 2]
```diff
@@ -377,79 +377,111 @@ func (c *JobClient) GetLastLogLineError(ctx context.Context, podNamespace, podNa
...
+			return logs, err
+		}
+
+		defer stream.Close()
+
+		buf := new(bytes.Buffer)
+		_, err = io.Copy(buf, stream)
+		if err != nil {
+			if len(logs) != 0 {
+				return logs, nil
```
Comment (vLia) [inline pkg/jobs/jobclient.go:421]: same here, should we make sure it's the right type of error before returning half-streamed logs with no error?

**Review by vLia — APPROVED** [review]
looks good 👍  had one minor question

File: pkg/jobs/jobclient.go (hunk 2 above)
Comment (vsukhin) [reply pkg/jobs/jobclient.go:421]: same approach

Comment (vsukhin) [reply pkg/jobs/jobclient.go:421]: https://github.com/kubernetes/test-infra/blob/d408ead6e01bf53681bb1758f5a4728b28628fdd/prow/cmd/deck/main.go#L1368

File: pkg/jobs/jobclient.go (hunk 1 above)
Comment (vLia) [reply pkg/jobs/jobclient.go:409]: Looks good enough! 👍

**Pull request merged**